### PR TITLE
fix: Improve remote evaluation fetch retry logic

### PR DESCRIPTION
**Pull request by tyiuhc**
### Summary

Improve remote evaluation fetch retry logic

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/experiment-go-server/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: pkg/experiment/types.go [hunk 1]
```diff
@@ -27,3 +29,12 @@ type Variant struct {
 	Key      string                 `json:"key,omitempty"`
 	Metadata map[string]interface{} `json:"metadata,omitempty"`
 }
+
+type FetchError struct {
```
Comment (bgiori) [inline pkg/experiment/types.go:33]: use internal `fetchError`

File: pkg/experiment/types.go [hunk 2]
```diff
@@ -27,3 +29,12 @@ type Variant struct {
...
+	StatusCode int
+	Message    string
+}
+
+func (e *FetchError) Error() string {
+	return fmt.Sprintf("Message: %s", e.Message)
+}
```
Comment (bgiori) [inline pkg/experiment/types.go:40]: What's this for? is this to implement the error interface? If so, golang error messages are almost always all lowercase :) minor nitpick.

**Review by bgiori — CHANGES_REQUESTED** [review]
how to test?

File: pkg/experiment/types.go (hunk 2 above)
Comment (tyiuhc) [reply pkg/experiment/types.go:40]: Yep, for the error interface!

**Review by bgiori — APPROVED** [review]

**Pull request merged**